frontend/src/features/admin-form/create/builder-and-design/useBuilderAndDesignStore.tsx

```diff
@@ -0,0 +1,96 @@
+import { isEqual } from 'lodash'
+import create from 'zustand'
+import { devtools } from 'zustand/middleware'
+
+import { FieldCreateDto, FormFieldDto } from '~shared/types/field'
+
+export enum BuildFieldState {
+  CreatingField,
+  EditingField,
+  Inactive,
+}
+
+export type BuilderAndDesignStore = {
+  fields: FormFieldDto[] | null
+  setFields: (fields: FormFieldDto[]) => void
+  updateCreateState: (field: FieldCreateDto, insertionIndex: number) => void
+  updateEditState: (field: FormFieldDto) => void
+  setToInactive: () => void
+  stateData:
+    | {
+        state: BuildFieldState.CreatingField
+        field: FieldCreateDto
+        insertionIndex: number
+      }
+    | {
+        state: BuildFieldState.EditingField
+        field: FormFieldDto
+      }
+    | { state: BuildFieldState.Inactive }
```

Comment:
yeap that was the initial implementation but I changed it to add an inactive state to make it more explicit. I think even if we allowed the state to be `null`, we would still need specific checks like the first one in your comment because there will likely be more states in future (`EditingStartPage` and `EditingEndPage`). so personally I'd rather keep it explicit to force these checks. wdyt?

Reply:
I'll defer to you!